### PR TITLE
chore(utils): Update internal service update input validation

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
Add more validation to the update input arguments to catch wrong usage 

FIXES TRI-64

### CODE REVIEW

**Comment by adrien2p** [comment]
I am also trying to improve the base API in a separate pr @olivermrbl, though, I ll have to ts expect error in many services because they override the base method with a different API which is not supposed to happen, but I can ignore them for now until the full modules API is settled and in any case those override normalize their input to be consumed by the base methods

File: packages/core/utils/src/modules-sdk/medusa-internal-service.ts [hunk 1]
```diff
@@ -252,12 +252,22 @@ export function MedusaInternalService<
       input: any | any[] | SelectorAndData | SelectorAndData[],
       @MedusaContext() sharedContext: Context = {}
     ): Promise<InferEntityType<TEntity> | InferEntityType<TEntity>[]> {
-      if (!isDefined(input) || (Array.isArray(input) && input.length === 0)) {
+      if (
```
Comment (carlos-r-l-rodrigues) [inline packages/core/utils/src/modules-sdk/medusa-internal-service.ts:282]: can we use `isPresent` here?

Comment (adrien2p) [reply packages/core/utils/src/modules-sdk/medusa-internal-service.ts:282]: yes you are right 👍 much better, I ll add that in

File: packages/core/utils/src/modules-sdk/medusa-internal-service.ts [hunk 2]
```diff
@@ -252,12 +252,22 @@ export function MedusaInternalService<
       input: any | any[] | SelectorAndData | SelectorAndData[],
```
Comment (olivermrbl) [inline packages/core/utils/src/modules-sdk/medusa-internal-service.ts:257]: thought: @shahednasser, do we have a reference for the internal module services? If not, would it make sense to have such that we can provide a link in the error message?

Comment (shahednasser) [reply packages/core/utils/src/modules-sdk/medusa-internal-service.ts:257]: Not a reference, but we have examples here: https://docs.medusajs.com/v2/advanced-development/modules/service-factory#generated-methods

We can generate a reference if we find it necessary, but the reason I didn't is I imagine the typings being confusing. We can alternatively have a document that provides more advanced / realistic examples

**Comment by adrien2p** [comment]
@olivermrbl I will stop this pr for now as it is still documented. But it require a much broader work on aligning the different API's the modules are defining VS what the internal service is provided. There is too much discrepancies for now and I d like to finalise my other tasks. wdyt?

thought, I ll keep this pr open for reference in case I get back to this one later

**Comment by olivermrbl** [comment]
> @olivermrbl I will stop this pr for now as it is still documented. But it require a much broader work on aligning the different API's the modules are defining VS what the internal service is provided. There is too much discrepancies for now and I d like to finalise my other tasks. wdyt?
> 
> thought, I ll keep this pr open for reference in case I get back to this one later

Yeah, let's do that for now. We can pick it up as part of our polishing phase.

**Pull request closed without merging**